src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -568,4 +600,48 @@ public boolean isFeedbackSessionForUserTypeToAnswer(FeedbackSessionAttributes se
         return session.isVisible() && !questionsToAnswer.isEmpty();
     }
 
+    private void updateFeedbackSessionsDeadlinesWithNewEmail(String courseId, String oldEmailAddress,
+            String newEmailAddress, boolean isInstructor) {
+        if (oldEmailAddress.equals(newEmailAddress)) {
+            return;
+        }
+        updateFeedbackSessionsDeadlinesForUser(courseId, oldEmailAddress, isInstructor,
+                deadlines -> deadlines.put(newEmailAddress, deadlines.remove(oldEmailAddress)));
+    }
+
+    private void deleteFeedbackSessionsDeadlinesForUser(String courseId, String emailAddress, boolean isInstructor) {
+        updateFeedbackSessionsDeadlinesForUser(courseId, emailAddress, isInstructor,
+                deadlines -> deadlines.remove(emailAddress));
+    }
+
+    private void updateFeedbackSessionsDeadlinesForUser(String courseId, String emailAddress, boolean isInstructor,
```

Comment:
This looks quite readable to me! Kudos!

Reply:
Kudos instead to Samuel lol, this is more of his code. I just modified his original suggestion a bit here and there.